bootstrap_cfn/config.py

```diff
@@ -264,29 +269,46 @@ def iam(self):
     def s3(self):
         # REQUIRED FIELDS AND MAPPING
         required_fields = {
-            'static-bucket-name': 'BucketName'
         }
 
-        # TEST FOR REQUIRED FIELDS AND EXIT IF MISSING ANY
-        present_keys = self.data['s3'].keys()
-        for i in required_fields.keys():
-            if i not in present_keys:
-                print "\n\n[ERROR] Missing S3 fields [%s]" % i
-                sys.exit(1)
-
-        bucket = Bucket(
-            "StaticBucket",
-            AccessControl="BucketOwnerFullControl",
-            BucketName=self.data['s3']['static-bucket-name'],
-        )
+        # As there are no required fields, although we may not have any
+        # subkeys we still need to be able to have a parent key 's3:' to
+        # signify that we want to create an s3 bucket. In this case we
+        # set up an empty (no options set) dictionary
+        if isinstance(self.data['s3'], dict):
+            present_keys = self.data['s3'].keys()
+            # TEST FOR REQUIRED FIELDS AND EXIT IF MISSING ANY
+            for i in required_fields.keys():
+                if i not in present_keys:
+                    print "\n\n[ERROR] Missing S3 fields [%s]" % i
+                    sys.exit(1)
+        else:
+            present_keys = {}
+
+        # If the static bucket name is manually set then use that,
+        # otherwise use the <stackname>-<logical-resource-name>-<random>
+        # default
+        if 'static-bucket-name' in present_keys:
+                bucket = Bucket(
+                    "StaticBucket",
+                    AccessControl="BucketOwnerFullControl",
+                    BucketName=self.data['s3']['static-bucket-name'],
+                )
+        else:
+            bucket = Bucket(
+                "StaticBucket",
+                AccessControl="BucketOwnerFullControl",
+            )
```

Comment:
You can simplify this too - you can set properties on the troposphere instances:

```python
bucket = Bucket(
    "StaticBucket",
    AccessControl="BucketOwnerFullControl",
)

if 'static-bucket-name' in present_keys:
    bucket.BucketName = self.data['s3']['static-bucket-name']
```